TableTennisGenerator/TableTennisScorer/Scorer.cs

```diff
@@ -1,11 +1,117 @@
+            _playerMetrics = new Dictionary<string, Dictionary<string, double>>();
+        }
+
+        public Dictionary<string, Dictionary<string, double>> GenerateMetrics()
+        {
+            using (StreamReader reader = new StreamReader(_filePath))
+            {
+                while (!reader.EndOfStream)
+                {
+                    string line = reader.ReadLine();
+                    string[] parts = line.Split(",");
+
+                    if (parts.Length == 8 && int.TryParse(parts[FIRST_SCORE], out int firstScore) && int.TryParse(parts[SECOND_SCORE], out int secondScore))
+                    {
+                        if (!_playerMetrics.ContainsKey(parts[FIRST_NAME]))
+                        {
+                            InitializePlayer(parts[FIRST_NAME]);
+                        }
+                        if (!_playerMetrics.ContainsKey(parts[FIRST_PARTNER_NAME]))
+                        {
+                            InitializePlayer(parts[FIRST_PARTNER_NAME]);
+                        }
+                        if (!_playerMetrics.ContainsKey(parts[SECOND_NAME]))
+                        {
+                            InitializePlayer(parts[SECOND_NAME]);
+                        }
+                        if (!_playerMetrics.ContainsKey(parts[SECOND_PARTNER_NAME]))
+                        {
+                            InitializePlayer(parts[SECOND_PARTNER_NAME]);
+                        }
+
+                        if (firstScore > secondScore)
+                        {
+                            SetMetrics(parts[FIRST_NAME], parts[FIRST_PARTNER_NAME], parts[SECOND_NAME], parts[SECOND_PARTNER_NAME], firstScore, secondScore);
+                        }
+                        else
+                        {
+                            SetMetrics(parts[SECOND_NAME], parts[SECOND_PARTNER_NAME], parts[FIRST_NAME], parts[FIRST_PARTNER_NAME], secondScore, firstScore);
+                        }
+                    }
+                }
+            }
+
+            foreach (KeyValuePair<string, Dictionary<string, double>> keyValuePair in _playerMetrics)
+            {
+                Dictionary<string, double> metrics = keyValuePair.Value;
+                metrics[PointsPerGame] = metrics[PointsScored] / metrics[GamesPlayed];
+                metrics[PointDifferential] = metrics[PointsScored] - metrics[PointsLost];
+            }
+
+            return _playerMetrics;
+        }
+
+        private void InitializePlayer(string player)
+        {
+            Dictionary<string, double> metrics = new Dictionary<string, double>();
+            metrics.Add(GamesPlayed, 0);
+            metrics.Add(GamesWon, 0);
+            metrics.Add(PointsScored, 0);
+            metrics.Add(PointsLost, 0);
+
+            _playerMetrics.Add(player, metrics);
+        }
+
+        private void SetMetrics(string player1, string partner1, string player2, string partner2, int winningScore, int losingScore)

```

Comment:
just for my sanity, could we use winner1, winner2 and loser1, loser 2 instead of player1, partner1, player2, partner2? Otherwise I keep getting confused and having to doublecheck whether winningScore is really always team1's score